### DISCUSSION TITLE
How to scan non-extension files ?

### DISCUSSION THREAD

**Discussion by Alceatraz** [immich-app/immich#9178] [Q&A]
Hello!

I'm using TrueNAS + immich 1.102.3, I'm pretty new to immich.

I have many source: Immich App backup / FolderSync backup / owncloud /  / smb upload / OSS, The problem occure with OSS: Scan external library will ignore non-extension files.

![1](https://github.com/immich-app/immich/assets/8266702/80fb16db-3311-4c58-be0f-3c3e500438e8)

Because of this is data folder, All filds are managed. So any rename or link will cause data corruption.

AFAK, Use [file magic number](https://en.wikipedia.org/wiki/List_of_file_signatures) can detect file type, And we don't care if a file is illegal ( just skip ).

So Question: Can immich scan non-extension file and detect file type? And how?

**Comment by huuthuan1310** [comment]
Same problem, anybody can help us?
Please!
